app/Database.hs

```diff
@@ -1,35 +1,116 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE InstanceSigs #-}
 
-module Database where
+module Database (getFrequencies, generateLines, getLines) where
 
+import Database.Models
 import Database.SQLite.Simple
+import qualified Data.Text as Text
+import Text.Printf
 
-data ActualArrival = ActualArrival { timestamp :: Int, stopId :: Int, punctuality :: Int, journeyId :: Int} deriving Show
+data TimeOfDay = TimeOfDay { hours :: Int, minutes :: Int }
 
-instance FromRow ActualArrival where
-    fromRow = ActualArrival <$> field <*> field <*> field <*> field
+instance Show TimeOfDay where
+    show :: TimeOfDay -> String
+    show (TimeOfDay h m) = printf "%02d" h ++ ":" ++ printf "%02d" m ++ ":00"
 
-data Frequency = Frequency { value :: Int, frequency :: Int} deriving Show
+data DayOfWeek = Mon | Tue | Wed | Thu | Fri | Sat | Sun
```

Comment:
I used `Data.Time.Calendar`'s `DayOfWeek` in the message format. It already has a Show instance, and you can use `weekToNumber = show . fromEnum`